### PR TITLE
Import error read endpoints and schema

### LINKED ISSUE #8130: API Endpoints - Read-only - Import errors
**Description**
Hello 

We need to create several endpoints that perform basic read-only operations on **Import error** . We need the following endpoints:

- GET /importErrors
- GET /importErrors/{import_error_id}

For now, we focus only on read-only operations, but others will also be implemented as the next step.

Detailed information is available in the issue:
https://github.com/apache/airflow/issues/8118

**Use case / motivation**
N/A

**Related Issues**
N/A

### PR DESCRIPTION
**Pull request by OmairK**
Closes #8130 
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/api_connexion/openapi/v1.yaml [hunk 1]
```diff
@@ -446,7 +446,7 @@ paths:
   /importErrors:
     get:
       summary: Get all import errors
-      operationId: airflow.api_connexion.endpoints.import_errror_endpoint.get_import_errors
+      operationId: airflow.api_connexion.endpoints.import_error_endpoint.get_import_errors
```
Comment (mik-laj) [inline airflow/api_connexion/openapi/v1.yaml:449]: 😻

File: tests/api_connexion/endpoints/test_import_error_endpoint.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+class TestBaseImportError(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        cls.app = app.create_app(testing=True)  # type:ignore
+
+    def setUp(self) -> None:
+        self.client = self.app.test_client()  # type:ignore
+        self.timestamp = "2020-06-10T12:02:44.870706"
```
Comment (mik-laj) [inline tests/api_connexion/endpoints/test_import_error_endpoint.py]: Can you reduce the precision of the dates given? Now it is difficult to compare them.

**Review by mik-laj — APPROVED** [review]

**Review by mik-laj — COMMENTED** [review]
We are very close to the end. The main problem is static checking.  As long as the change is not green (excluding quarantine tests), I cannot merge it. 
Here is a guide about static checks: https://github.com/apache/airflow/blob/master/STATIC_CODE_CHECKS.rst
Can you do a rebase also? We have a merge conflicts.

**Comment by mik-laj** [comment]
```
tests/api_connexion/endpoints/test_import_error_endpoint.py::TestGetImportErrorEndpoint::test_response_200
FAILED tests/api_connexion/endpoints/test_import_error_endpoint.py::TestGetImportErrorsEndpoint::test_get_import_errors
FAILED tests/api_connexion/endpoints/test_import_error_endpoint.py::TestGetImportErrorsEndpointPagination::test_limit_and_offset_0__api_v1_importErrors_limit_1
FAILED tests/api_connexion/endpoints/test_import_error_endpoint.py::TestGetImportErrorsEndpointPagination::test_limit_and_offset_1__api_v1_importErrors_limit_110
FAILED tests/api_connexion/endpoints/test_import_error_endpoint.py::TestGetImportErrorsEndpointPagination::test_limit_and_offset_2__api_v1_importErrors_offset_1
FAILED tests/api_connexion/endpoints/test_import_error_endpoint.py::TestGetImportErrorsEndpointPagination::test_limit_and_offset_3__api_v1_importErrors_offset_3
FAILED tests/api_connexion/endpoints/test_import_error_endpoint.py::TestGetImportErrorsEndpointPagination::test_limit_and_offset_4__api_v1_importErrors_offset_3_limit_3
```
Some API-related tests were unsuccessful. Could you look at it?

**Comment by OmairK** [comment]
> ```
> tests/api_connexion/endpoints/test_import_error_endpoint.py::TestGetImportErrorEndpoint::test_response_200
> FAILED tests/api_connexion/endpoints/test_import_error_endpoint.py::TestGetImportErrorsEndpoint::test_get_import_errors
> FAILED tests/api_connexion/endpoints/test_import_error_endpoint.py::TestGetImportErrorsEndpointPagination::test_limit_and_offset_0__api_v1_importErrors_limit_1
> FAILED tests/api_connexion/endpoints/test_import_error_endpoint.py::TestGetImportErrorsEndpointPagination::test_limit_and_offset_1__api_v1_importErrors_limit_110
> FAILED tests/api_connexion/endpoints/test_import_error_endpoint.py::TestGetImportErrorsEndpointPagination::test_limit_and_offset_2__api_v1_importErrors_offset_1
> FAILED tests/api_connexion/endpoints/test_import_error_endpoint.py::TestGetImportErrorsEndpointPagination::test_limit_and_offset_3__api_v1_importErrors_offset_3
> FAILED tests/api_connexion/endpoints/test_import_error_endpoint.py::TestGetImportErrorsEndpointPagination::test_limit_and_offset_4__api_v1_importErrors_offset_3_limit_3
> ```
> 
> Some API-related tests were unsuccessful. Could you look at it?

That's weird, these tests are running fine in my breeze environment(both python3.6 python 3.7)

**Comment by mik-laj** [comment]
@OmairK Okay. I looked at what the problem was. The problem is that `clear_db_import_errors` function doesn't reset the sequence.  Each newly added element has a new identifier. However, if the object was created in another test, then the numbering continues in your test. If you run one test several times, only the first run will be correct, but all others will fail. I have prepared a patch that normalizes identifiers in responses.
https://github.com/apache/airflow/pull/9217/commits

**Comment by OmairK** [comment]
> @OmairK Okay. I looked at what the problem was. The problem is that `clear_db_import_errors` function doesn't reset the sequence. Each newly added element has a new identifier. However, if the object was created in another test, then the numbering continues in your test. If you run one test several times, only the first run will be correct, but all others will fail. I have prepared a patch that normalizes identifiers in responses.
> https://github.com/apache/airflow/pull/9217/commits
> This should help. What do you think about this?

Ah it was the sequence all along, I get it thank you so much for the help. A couple of quick questions

1. I was running these tests locally by `./breeze test-target tests/api_connexion/` these tests should have failed there too unless the sequence was automatically resetting after every test, which doesn't make sense. Is there some major difference between how breeze and github action CI handles the db state between tests ?
2. If yes, how do you reenact your airflow github actions locally ? (This would help me a lot in the future) 

P.S.: +1 for the commit message, it perfectly sums up the situation :)

**Comment by mik-laj** [comment]
1. I will reproduce this problem locally. I launched Breeze with Postgres and Python 3.8 and have run the following command:
```
pytest tests/api_connexion/endpoints/test_import_error_endpoint.py
```
I suspect you use SQLite, which sometimes behaves differently

**Comment by mik-laj** [comment]
Caa you do a rebase to the latest master?

Is this still WIP? There is still a label in the PR title.

**Comment by OmairK** [comment]
> Is this still WIP? There is still a label in the PR title.

No, I had 2 PRs to be merged at the same time so I made this PR WIP just to keep track.

**Pull request merged**